packages/sdk-components-react/src/theme/tailwind-theme.ts

```diff
@@ -0,0 +1,27 @@
+// eslint-disable-next-line import/no-internal-modules
+import config from "tailwindcss/defaultConfig";
+
+type Theme = NonNullable<(typeof config)["theme"]>;
+
+type ThemeProperties = keyof Theme;
+
+type Extract<K> = NonNullable<K> extends infer X
+  ? X extends (...args: never[]) => infer T
+    ? T
+    : X
+  : K;
+
+type ThemeResolved = {
+  [key in keyof Theme]: Extract<Theme[key]>;
+};
+
+export const theme = <T extends ThemeProperties>(name: T): ThemeResolved[T] => {
+  const value = config?.theme?.[name];
+
+  if (typeof value === "function") {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-return
+    return value({ theme, colors: {} });
```

Comment:
colors is not part of the theme? is it the same in tw?

Reply:
Part of, I just didnt include to avoid hsl etc parsing

Reply:
Gonna hoist config, in that case can extend it better

Reply:
if you change the theme spec its gonna be confusing over time

Reply:
Added some simle theme colors etc override